server/lookup/faiss.py

```diff
@@ -37,6 +43,72 @@ def search(self, data, k):
                 results.append({"user_id": self.user_ids[i], "distance": float(d)})
         return results
 
+    def known_users(self):
+        return self.user_ids
+
+    def evaluate(self, data, print_info=False):
+        y_true = []
+        y_pred = []
+        for _, i in data.iterrows():
+
+            df = pd.DataFrame(i.to_dict(), index=["user_id"])
+            try:
+                df = df.drop("_id", axis="columns")
+            except KeyError:
+                pass
+            try:
+                df = df.drop("user_id", axis="columns")
+            except KeyError:
+                pass
+            result = self.search(df.to_numpy("float32"), config["NEIGHBOURS"])
+            if len(result) > 0:
+                if i["user_id"] in self.known_users():
+                    y_true.append(i["user_id"])
+                else:
+                    y_true.append("newUser")
+
+                if result[0]["distance"] <= config["MAXIMAL_DISTANCE"]:
```

Comment:
Nebolo by lepsie keby tuto logiku rovno robil `search`?

Reply:
áno, mal by, ale to nebola úplne náplň tohoto tasku... to sa tam malo asi zaradiť niekedy po dokončení toho thresholdu čo robil @matuskovac, ja som tu len prevzal vyhodnocovanie modelu...